File: booking-app/src/app/models/accommodation.ts
import { AccommodationType } from './accommodation-type';
import { User } from './user';
import { Place } from './place';
import { Room } from './room';
export class Accommodation {
    Id: number;
    Name: string;
    Description: string;
    Address: string;
    AvrageGrade: number;
    Latitude: number;
    Longitude: number;
    ImageUrl: string;
    Approved: boolean;
    PlaceId: number;
    AccommodationTypeId: number;
    OwnerId: number;
    Place: Place;
    Rooms: Room[];
    Owner: User;
    AccommodationType: AccommodationType;

      constructor(Id?: number, Name?: string, Description?: string, Address?: string, Approved? : boolean, 
        ImageUrl? : string, Latitude?: number, Longitude?: number, PlaceId?: number, AccommodationTypeId?: number, OwnerId?: number
        ) {

      this.Id = Id;
      this.Name = Name;
      this.Description = Description;
      this.Address = Address;
      this.Latitude = Latitude;
      this.Longitude = Longitude;
      this.Rooms = [];
      this.Approved = Approved;
      this.ImageUrl = ImageUrl;
      this.AccommodationType = new AccommodationType;
      this.Place = new Place();
      this.PlaceId = PlaceId;
      this.AccommodationTypeId = AccommodationTypeId;
      this.OwnerId = OwnerId;
    }
}

File: BookingApp/Models/Data/Accommodation.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace BookingApp.Models.Data
{
    public class Accommodation
    {
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Name { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        [StringLength(50)]
        public string Address { get; set; }

        [Range(0, 5)]
        public double AverageGrade { get; set; }

        [Required]
        public double Latitude { get; set; }

        [Required]
        public double Longitude { get; set; }

        public string ImageURL { get; set; }

        [Required]
        public bool Approved { get; set; }

        [Required]
        [ForeignKey("AccommodationType")]
        public int AccommodationTypeId { get; set; }

        public AccommodationType AccommodationType { get; set; }

        [Required]
        [ForeignKey("Place")]
        public int PlaceId { get; set; }

        public Place Place { get; set; }

        public List<Comment> Comments { get; set; }
        public List<Room> Rooms { get; set; }
        [Required]
        [ForeignKey("Owner")]
        public int OwnerId { get; set; }

        public AppUser Owner { get; set; }
    }
}
